src/main/scala/no/ndla/learningpathapi/service/UpdateService.scala

```diff
@@ -463,6 +469,29 @@ trait UpdateServiceComponent {
       accessGranted(learningPath, owner)
     }
 
+    private def getEditableLearningPath(learningPathId: Long,
```

Comment:
Usikker på om jeg liker denne. Litt avhengig av hvilke rettigheter en pulisher skal ha (og at jeg har forstått koden riktig :p) tror jeg det er et hull her. Siden dette endepunktet kalles fra `PATCH /:learningpath_id` og `PUT /:learningpath_id/status/` og `DELETE /:learningpath_id` har i praksis publishers mulighet for å oppdatere alle felter i alles læringsstier. I tillegg har de også mulighet for å kunne sette en vilkårlig status på alles læringsstier.

Jeg foreslår å refaktorere her litt siden vi nå har både `withIdAndAccessGranted` og `getEditableLearningPath`. Det er ikke åpenbart hvilke av de to funksjonene man skal bruke i hvilke tilfeller. Og hvis vi skal få en ny rolle `admin` snart blir det kanskje enda vanskeligere.
Hva om vi dropper både denne funksjonen og `withIdAndAccessGranted` og innfører nye funksjoner som sjekker om man har tilgang til å gjøre spesifikke ting. En egen case class med bruker-info kunne kanskje også vært nyttig. Den kunne kanskje sett sånn ut: `case class UserInfo(userId: String, roles: Set[LearningpathRole.Value])`. Kallet til `getEditableLearningpath` kan da kanskje bli sånn:
`withId(lpId, includeDeleted=true).canSetStatus(status, userInfo)`. Kallet til `withIdAndAccessGranted` kunne sett sånn ut: `withId(lpId).canEditLearningpath(userInfo)`.

Reply:
Lager admin istedet og refaktorerer ca som du sa :) :partly_sunny: